### ISSUE TITLE
Panpsychan to treat overdoses - research for breakthrough MKII

### ISSUE THREAD

**Issue by I-Knight-I** [B-Knight49/Rimworld-MedicalOverhaul#53]
Need to also categorise each drug (where possible) into opiates or others like amphetamines. Need to make this explicitly clear to the player too.

**Comment by ghost** [comment]
Note that Narcan is only typically used to treat opiate/opioid overdoses.  It functions by blocking the opioid receptors on your body's cells.  Since the receptors are blocked by the Narcan, the opioids cannot attach to them and enact their effects.

AFAIK the best way to treat opiate/opioid addictions is to wean them off of it.  There is a medication called methadone available which is essentially used as a substitution therapy until the patient is weaned off enough to quit without severe withdrawal effects.

**Comment by I-Knight-I** [comment]
My mistake again - I meant "overdoses" instead of "addictions". Editing now. 

Since narcan essentially stops opiates from working we could make it so that there is a cooldown between taking the anti-drug and then any opioids shortly after. Could be a balanced way to prevent players from carrying a lot of a performance enhancing drugs and narcan for any overdoses that'll cause (basically allowing for constant performance boosts). 

Could consider adding Methadone too. Would make it lower the pawns dependency on opiates whilst not giving them any enhanced health benefits that the drugs would otherwise offer. This'd act as the drug they're addicted to (thus removing any withdrawals) and it'd reduce their "dependency" hediff too.

**Comment by ghost** [comment]
Regarding the way Narcan works to prevent opioids from working: although it
prevents them from acting on the cells, the opioids can still remain in the
bloodstream after the Narcan wears off.  This is why it's important for
people who are rescued by Narcan not to use the opioids again immediately
after recovery.  The Narcan prevents the opioids from working, but they're
still in the bloodstream.  If you use more opioids, you're adding more to
the bloodstream, so when the Narcan wears off suddenly all of those opioids
take effect.

**Comment by I-Knight-I** [comment]
Heh, yeah. Unfortunately some players will use that to their advantage so it might be a clever idea to simply make the narcan hediff last dramatically longer than the effects of any in-game opiate. That way we can guarantee that it isn't exploited by ensuring that current opiates the pawn has consumed wear off and that any consumed during the narcan hediff wastes resources as it is not effective. It also gives a period for the pawn to 'come down' so that no more drugs can be rushed mid-battle. 

We can make it simpler so that, when the pawn has the narcan hediff, any opioid hediff is removed instantly - existing or not. From there the pawn is forced to 'come down' by the narcan lasting for a long time (maybe an in-game day?).

**Comment by I-Knight-I** [comment]
Finished main functionality and completed code in 3c2078c. 

EDIT: Nevermind. Drug Overdose hediff isn't reduced. Not sure how to implement this to only work for opiate overdoses?

**Comment by I-Knight-I** [comment]
Fixed the above. I've now checked if the pawn has any opiate hediff removed by the Narcan and, if they do, then to check if they're overdosing. If they are then lower the severity of the hediff by 0.5. This means that overdoses caused by any drug that's not an opiate won't work. 

~~Minor bug: if a pawn combines both an opiate AND a non-opiate in order to overdose then the OD hediff will still be reduced by 0.5. I think this isn't a large problem, though, because the likelihood of a pawn not dying from two different hard drugs is slim.~~

Commit: a8c9935

Fixed bug in commit: 47921f3
